src/main/java/ch/njol/skript/aliases/ItemData.java

```diff
@@ -247,13 +254,55 @@ public String toString() {
 		return toString(false, false);
 	}
 	
-	public String toString(final boolean debug, final boolean plural) {
+	public String toString(boolean debug, boolean plural) {
 		StringBuilder builder = new StringBuilder(Aliases.getMaterialName(this, plural));
 		ItemMeta meta = stack.getItemMeta();
-		if (meta != null && meta.hasDisplayName()) {
-			builder.append(" ").append(m_named).append(" ");
-			builder.append(meta.getDisplayName());
+
+		Map<Enchantment, Integer> enchantments = stack.getEnchantments();
+		if (!enchantments.isEmpty()) {
+			builder.append(Language.getSpaced("enchantments.of").toLowerCase(Locale.ENGLISH));
+			int i = 0;
+			for (Entry<Enchantment, Integer> entry : enchantments.entrySet()) {
+				if (i != 0) {
+					if (i != enchantments.size() - 1) {
+						builder.append(", ");
+					} else  {
+						builder.append(" ").append(GeneralWords.and).append(" ");
+					}
+				}
+				Enchantment ench = entry.getKey();
+				if (ench == null)
+					continue;
+				builder.append(EnchantmentType.toString(ench));
+				builder.append(" ");
+				builder.append(entry.getValue());
+				i++;
+			}
 		}
+
+		if (meta != null) {
+			if (meta.hasDisplayName()) {
+				builder.append(" ").append(m_named).append(" ");
+				builder.append("\"").append(SkriptColor.replaceColorChar(meta.getDisplayName())).append("\"");
```

Comment:
I'm not sure we should color this by default. IMO it should be up to the user to color the string.